### ISSUE TITLE
Waiting for database server to accept connections. Tag 7.10.1

### ISSUE THREAD

**Issue by rawadhilal88** [sameersbn/docker-gitlab#306]
Hello,
  I keep getting 'Waiting for database server to accept connections........' when I try running the gitlab latest image 7.10.1. I'm linking it with mysql container.
The following is what I'm using: <code><pre>docker run --name=gitlab --link mysql:mysql --link redis:redisio \
    -e "DB_USER=user" -e "DB_PASS=passowd" \
    -e "DB_NAME=db" \
    -e "GITLAB_HOST=git.local.host" \
    -v /opt/gitlab/data:/home/git/data \
    -p 10022:22 -p 10080:80 \
    sameersbn/gitlab:latest
</pre></code>

But if I run: <code><pre>docker run --name=gitlab --link mysql:mysql --link redis:redisio \
    -e "DB_USER=user" -e "DB_PASS=passowd" \
    -e "DB_NAME=db" \
    -e "GITLAB_HOST=git.local.host" \
    -v /opt/gitlab/data:/home/git/data \
    -p 10022:22 -p 10080:80 \
    --rm -it sameersbn/gitlab:latest /bin/bash
</pre></code>

and then from bash run the: <code><pre>sudo -u git -H bundle exec rake gitlab:env:info RAILS_ENV=production
</pre></code>
The call will return no errors. The database is accessible from the gitlab container. I'm not sure why its not working.

**Comment by sameersbn** [comment]
@rawadhilal88 maybe the DB_USER is not allowed remote access to the database. From the gitlab container try connecting to the mysql server using the `mysql` command. eg. `mysql -h <host> -u <user> -p <password>`

**Comment by rawadhilal88** [comment]
@sameersbn Hi Mr. Sameer. There was nothing wrong with the connection. I was able to connect properly. However, the problem lies in a password character that was causing the issue. For some reason, the script does not like the character '!' to be in the password. For example:
<code><pre>docker run --name=gitlab --link mysql:mysql --link redis:redisio \
    -e "DB_USER=user" -e "DB_PASS=pas!sowd" \
    -e "DB_NAME=db" \
    -e "GITLAB_HOST=git.local.host" \
    -v /opt/gitlab/data:/home/git/data \
    -p 10022:22 -p 10080:80 \
    sameersbn/gitlab:latest
</pre></code>

I fixed it by changing my password. Thanks for your help anyways. :)
